Source/WarpX.cpp

```diff
@@ -280,8 +285,10 @@ WarpX::WarpX ()
 
     // Allocate field solver objects
 #ifdef WARPX_USE_PSATD
-    spectral_solver_fp.resize(nlevs_max);
-    spectral_solver_cp.resize(nlevs_max);
+    if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
+        spectral_solver_fp.resize(nlevs_max);
+        spectral_solver_cp.resize(nlevs_max);
+    }
 #endif
     m_fdtd_solver_fp.resize(nlevs_max);
```

Comment:
It's a small thing, but do these need to be resized when using PSATD?

Reply:
Yes, I was wondering as well and just kept to keep the logic change small. We could guard this.